adala/runtimes/_openai.py

```diff
@@ -363,10 +363,15 @@ async def batch_to_batch(
                 )
 
                 # parse responses, optionally match it with options
-                for prompt, response in zip(prompts, responses):
+                index = []
+                for prompt, response, idx in zip(prompts, responses, batch.index):
                     # check for errors - if any, append to outputs and continue
                     if response.get("error"):
-                        outputs.append(response)
+                        # FIXME if we collect failed and succeeded outputs in the same list -> df, we end up with an awkward schema like this:
```

Comment:
Also, can we explicitly state here that we're intentionally not appending to `outputs` here? I know it's clear in context, but I could still see someone thinking this was a mistake if we don't explicitly explain